### PR TITLE
impl(internal/parser): merge protobuf and openapi packages

### PR DESCRIPTION
**Pull request by julieqiu**
Merge protobuf and openapi to a single package parser. This makes it easier to share code between these packages in future PRs.

### CODE REVIEW

File: generator/internal/parser/openapi.go [hunk 1]
```diff
@@ -12,9 +12,9 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-// Package openapi reads OpenAPI v3 specifications and converts them into
+// Package praser reads specifications and converts them into
```
Comment (coryan) [inline generator/internal/parser/openapi.go:15]: s/praser/parser/

File: generator/internal/parser/protobuf_test.go [hunk 2]
```diff
@@ -1025,7 +1025,7 @@ func newTestCodeGeneratorRequest(t *testing.T, filename string) *pluginpb.CodeGe
 	return request
 }
 
-func checkMessage(t *testing.T, got genclient.Message, want genclient.Message) {
+func checkProtobufMessage(t *testing.T, got genclient.Message, want genclient.Message) {
```
Comment (coryan) [inline generator/internal/parser/protobuf_test.go:1028]: I am sure you noticed, but I think these helpers and the OpenAPI counterparts could be merged now.

**Review by coryan — APPROVED** [review]
One little typo, otherwise LGTM.

File: generator/internal/parser/protobuf_test.go (hunk 2 above)
Comment (julieqiu) [reply generator/internal/parser/protobuf_test.go:1028]: Ah I missed this! Will merge in a follow up.

**Pull request merged**